OpenRA.Game/Network/UnitOrders.cs

```diff
@@ -70,6 +70,19 @@ internal static void ProcessOrder(OrderManager orderManager, World world, int cl
 							TextNotificationsManager.AddPlayerJoinedLine(localizedMessage.Key, localizedMessage.Arguments);
 						else if (localizedMessage.Key == Left)
 							TextNotificationsManager.AddPlayerLeftLine(localizedMessage.Key, localizedMessage.Arguments);
+						else if (localizedMessage.Key.Contains("briefing"))
```

Comment:
This doesn't look very safe?!

Reply:
Suggestions?

Reply:
All messages should be routed through the MapPreview. This is also the case for lobby options, faction and bot names.

Reply:
It is already routed through `mapPreview.GetLocalisedString(localizedMessage.Key)`. Do you mean `public string Briefing` should be a field in `MapPreview`?

Reply:
I think the issue is that the server sends the map to the client in this message. The client should already have the map available. It also seems like the place the server sends the message has a HACK flag attached.

https://github.com/OpenRA/OpenRA/blob/5f59389359b67f9c0a929966cbf621ea4fbb2ab1/OpenRA.Mods.Common/ServerTraits/LobbyCommands.cs#L1464-L1465

Ideally I imagine we should not be involving the server at all. The briefing should show up when the client loads the map preview. I'd imagine it being called in `LobbyLogic.UpdateCurrentMap`. Though this may end up messing up the message order?

The other solution could be to send the briefing message only after making sure lobby options are synced. It looks like the sync order is sent right after the we send the briefing command

Reply:
Currently the very first message is the briefing. So if it were to become last it wouldn't be an issue

Reply:
@pchote does it make sense to have briefing posted client side?

Reply:
Removed the hacky mission briefing dump into lobby chat so it does not cause any more holdups.

Reply:
We still need to show the briefing in skirmish lobby, we can't just axe it. I do agree that dumping it in chat was not a good way to display it.

Reply:
That is also scope creep ;)

Reply:
you are removing a feature with no plan to replace it. While it is a major inconvenience it's not scope screep

Reply:
we don't need a fully fledged feature for next release, a hack that dumps the briefing into the chat can be re-added, just in a different place

Reply:
I removed it on purpose so we can move forward faster.